source/opt/debug_info_manager.cpp

```diff
@@ -544,9 +605,24 @@ uint32_t DebugInfoManager::GetVariableIdOfDebugValueUsedForDeclare(
   auto* operation = GetDbgInst(
       expr->GetSingleWordOperand(kDebugExpressOperandOperationIndex));
   if (operation == nullptr) return 0;
-  if (operation->GetSingleWordOperand(kDebugOperationOperandOperationIndex) !=
-      OpenCLDebugInfo100Deref) {
-    return 0;
+
+  // OpenCL.DebugInfo.100 contains a literal for the operation, Vulkan uses an
+  // OpConstant.
+  if (inst->IsOpenCL100DebugInstr()) {
+    if (operation->GetSingleWordOperand(kDebugOperationOperandOperationIndex) !=
+        OpenCLDebugInfo100Deref) {
+      return 0;
+    }
+  } else {
+    const analysis::Constant* operation_const =
+        context()->get_constant_mgr()->GetConstantFromInst(
+            context()->get_def_use_mgr()->GetDef(
+                operation->GetSingleWordOperand(
+                    kDebugOperationOperandOperationIndex)));
```

Comment:
I found it is used in three places.
Could you please create a function for
```
        context()->get_constant_mgr()->GetConstantFromInst(
            context()->get_def_use_mgr()->GetDef(
                operation->GetSingleWordOperand(
                    kDebugOperationOperandOperationIndex)));
```
?

Reply:
Once we have all changes integrated I plan to start adding regression tests. They will change too much and create a large review burden until that time. I will keep processing of the nonsemantic debuginfo disabled until regression tests are in place.

Reply:
Ok. Make sure we need unit tests not only for each optimization pass but also for `DebugInfoManager` (see https://github.com/KhronosGroup/SPIRV-Tools/blob/master/test/opt/debug_info_manager_test.cpp).
I am sorry if it is too much workload but I cannot believe the code updates without unit tests for them.

Reply:
> I am sorry if it is too much workload but I cannot believe the code updates without unit tests for them.

I understand your concern, however, please know that the branch I am cherry-picking from was able to "legalize" all but a couple of Sascha Willems HLSL shaders compiled with dxc -fspv-debug=vulkan and create valid SPIR-V.

I will try to put a few unit tests in with the next PR which will enable support the nonsemantic debuginfo to give you more confidence going forward.

Reply:
Ok. Thanks. Could you please address other comments I added in this PR?
Let me know my comments do not make sense.

Reply:
OK. Function now called.